File: app/src/main/java/za/co/duartbreedt/androidconceptsreference/pusheen/PusheenDetailsActivity.java
package za.co.duartbreedt.androidconceptsreference.pusheen;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;

import java.util.ArrayList;
import java.util.List;

import za.co.duartbreedt.androidconceptsreference.databinding.ActivityPusheenDetailsBinding;

public class PusheenDetailsActivity extends AppCompatActivity {

    private ActivityPusheenDetailsBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityPusheenDetailsBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
    }

    @Override
    protected void onStart() {
        super.onStart();

        List<String> pusheenNames = new ArrayList<String>(){{
            add("Vampire Pusheen");
            add("Donut Pusheen");
            add("Love Bird Pusheen");
            add("Snac Pusheen");
            add("Mask Pusheen");
            add("Summer Pusheen");
            add("Telephone Pusheen");
            add("Cake Pusheen");
            add("Scooter Pusheen");
            add("Relax Pusheen");
            add("Posh Pusheen");
            add("UFO Pusheen");
            add("Nessie Pusheen");
            add("Nessie Pusheen");
            add("Paris Pusheen");
            add("Snuggle Pusheen");
            add("Father Pusheen");
            add("Pudding Pusheen");
            add("Unicorn Pusheen");
            add("Ducky Pusheen");
            add("Bowtie Pusheen");
            add("Knitting Pusheen");
            add("Fishing Pusheen");
            add("Harry Pusheen");
            add("McDonalds Pusheen");
            add("Pooping Pusheen");
            add("Pizza Pusheen");
            add("Christmas Sweater Pusheen");
            add("Smors Pusheen");
            add("Dragon Pusheen");
            add("Icecream Pusheen");
        }};

        setupPusheenRecyclerView(pusheenNames);
    }

    private void setupPusheenRecyclerView(List<String> pusheenNames) {
        PusheenAdapter adapter = new PusheenAdapter(pusheenNames);
        binding.pusheenRecyclerView.setAdapter(adapter);

        LinearLayoutManager layoutManager = new LinearLayoutManager(getApplicationContext(), LinearLayoutManager.VERTICAL, false);
        binding.pusheenRecyclerView.setLayoutManager(layoutManager);
    }
}

File: app/src/main/java/za/co/duartbreedt/androidconceptsreference/pusheen/PusheenDetailsActivity.kt
package za.co.duartbreedt.androidconceptsreference.pusheen

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import za.co.duartbreedt.androidconceptsreference.databinding.ActivityPusheenDetailsBinding

class PusheenDetailsActivity : AppCompatActivity() {

    private var binding: ActivityPusheenDetailsBinding? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityPusheenDetailsBinding.inflate(layoutInflater)

        setContentView(binding?.root)
    }

    override fun onStart() {
        super.onStart()

        val pusheenNames: List<String> = listOf(
            "Vampire Pusheen",
            "Donut Pusheen",
            "Love Bird Pusheen",
            "Snac Pusheen",
            "Mask Pusheen",
            "Summer Pusheen",
            "Telephone Pusheen",
            "Cake Pusheen",
            "Scooter Pusheen",
            "Relax Pusheen",
            "Posh Pusheen",
            "UFO Pusheen",
            "Nessie Pusheen",
            "Nessie Pusheen",
            "Paris Pusheen",
            "Snuggle Pusheen",
            "Father Pusheen",
            "Pudding Pusheen",
            "Unicorn Pusheen",
            "Ducky Pusheen",
            "Bowtie Pusheen",
            "Knitting Pusheen",
            "Fishing Pusheen",
            "Harry Pusheen",
            "McDonalds Pusheen",
            "Pooping Pusheen",
            "Pizza Pusheen",
            "Christmas Sweater Pusheen",
            "Smors Pusheen",
            "Dragon Pusheen",
            "Icecream Pusheen"
        )

        setupPusheenRecyclerView(pusheenNames)
    }

    override fun onDestroy() {
        super.onDestroy()

        binding = null
    }

    private fun setupPusheenRecyclerView(pusheenNames: List<String>) {

        val adapter = PusheenAdapter(pusheenNames)

        binding?.pusheenRecyclerView?.adapter = adapter

        val layoutManager =
            LinearLayoutManager(applicationContext, LinearLayoutManager.VERTICAL, false)

        binding?.pusheenRecyclerView?.layoutManager = layoutManager
    }
}